Vizzuality/marxan-cloud
# nestjs base service integration

### Overview

Integration of the Vizzuality NestJS BaseService - linking this up to the core entities

### Designs

N/A

### Testing instructions

Make sure all the envvars listed in the documentation/`env.default` are defined in the `.env` file, then:

```
make stop
make clean-slate
make start
```

The API service should build and start correctly, all the migrations should run successfully (there should be 5 of them).

```
make seed-api-with-test-data
```

Should successfully import seed data. It should now be possible to obtain a token for one of the test users, e.g.

```
curl -X POST -H 'Content-Type: application/json' http://localhost:3030/auth/login -d '{"username": "aa@example.com", "password": "aauserpassword"}'
```

Most of the API endpoints listed on the OpenAPI documentation (e.g. http://localhost:3030/swagger) should work, after supplying the token obtained at the previous step. Most of the endpoints only return mock data. Some endpoints actually interact with the API db, but this will be the focus of a next PR.

### Feature relevant tickets

https://www.pivotaltracker.com/story/show/176552508
https://www.pivotaltracker.com/story/show/176376639

---

## Checklist before submitting

      deploying to staging/production, please add brief testing instructions
      to the deploy checklist (`docs/deployment-checklist.md`)
